forge-core/src/main/kotlin/io/github/hellocuriosity/providers/StringProvider.kt

```diff
@@ -20,9 +21,8 @@ class StringProvider(
     }
 
     override fun get(): String {
-        if (wordCount <= 0) {
-            println("$wordCount is not a valid input, reverting to default $DEFAULT_VALUE")
-            return lorem.getWords(wordCount)
+        if (wordCount < 0) {
```

Comment:
Let's make use of kotlin's `when` expression, so that we can easily add a new condition if necessary. Something like:

```
when {
    wordCount < 0 -> throw ModelForgeException("Cannot generate words of count $wordCount")
    else -> lorem.getWords(wordCount)
}
```

I'd also make the error message more explicit to what the library is doing, like in the example above. The forge can't `generate` a word from a negative `wordCount`

I know that the `lorem` library returns an empty string if it gets passed a `0`, but we should be more explicit here and handle that scenario ourselves, so that changes to that library don't make functional changes to ours. This will also remove the side effect from this function. This is easily done, by adding another condition to the when statement 👍